class-components/src/components/ErrorBoundary/ErrorBoundary.tsx

```diff
@@ -0,0 +1,27 @@
+import { Component, ReactNode, ErrorInfo } from "react";
+
+interface ErrorBoundaryProps {
+  children: ReactNode;
+}
+interface ErrorBoundaryState {
+  hasError: boolean;
+}
+
+export class ErrorBoundary extends Component<ErrorBoundaryProps, ErrorBoundaryState> {
+  state = { hasError: false };
+
+  static getDerivedStateFromError(): ErrorBoundaryState | undefined {
```

Comment:
```suggestion
  static getDerivedStateFromError() {
```
тебе не обязательно типизировать методы. можешь не тратить на это время. TS сам ругнется если где-то что-то надо будет явно типизировать, как, например, аргументы в функциях

Reply:
Понял.